.github/workflows/build-and-deploy.yml

```diff
@@ -51,17 +51,32 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
-      - name: Docker image tag
-        id: image
-        run: |
-          echo ::set-output name=tag::$CONTAINER_REGISTRY/$(echo $GITHUB_REPOSITORY | tr '[:upper:]' '[:lower:]'):$GITHUB_SHA
-        env:
-          CONTAINER_REGISTRY: ${{ env.CONTAINER_REGISTRY }}
-        shell: bash
-
       - uses: ./.github/workflows/actions/deploy-environment
         id: deploy
         with:
           environment_name: ${{ matrix.environment }}
           image_tag: ${{ github.sha }}
           azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+
+  deploy_production:
+    name: Deploy to production environment
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main' && github.event_name == 'push'
+    environment:
+      name: production
+      url: ${{ steps.deploy.outputs.environment_url }}
+    concurrency: deploy_production
+    needs: [docker, deploy_nonprod]
```

Comment:
Do we have to depend on docker?

Reply:
I will change that to deploy_nonprod only